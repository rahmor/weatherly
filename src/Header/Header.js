import React from 'react';
import './Header.css';

function Header() {
  return (
    <header className='Header-container'>
      <h1 className='Header-title'>Weatherly</h1>
      <form>
        <input type='search' placeholder='Search by City' />
        <button type='submit'>Search</button>
      </form>
      <h2>Atlanta, Ga</h2>
    </header>
  );
}

export default Header;
