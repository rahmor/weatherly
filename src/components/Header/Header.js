import React from 'react';
import { connect } from 'react-redux';
import './Header.css';
import submitCity from '../../actions/actions';

class Header extends React.Component {
  state = {
    city: '',
  };

  updateCity = (event) => {
    this.setState({ city: event.target.value });
  };

  submitCity = (event) => {
    event.preventDefault();
    this.props.dispatch(submitCity(this.state.city));
  };

  render() {
    return (
      <header className='Header-container'>
        <h1 className='Header-title'>Weatherly</h1>
        <form onSubmit={this.submitCity}>
          <label className='App-visually-hidden' htmlFor='location'>
            Search by City
          </label>
          <input
            id='location'
            type='search'
            name='location'
            placeholder='Search by City'
            value={this.state.value}
            onChange={this.updateCity}
          />
          <button type='submit' name='submitCity'>
            Search
          </button>
        </form>
        <h2>{this.props.city}</h2>
      </header>
    );
  }
}

export default connect()(Header);
