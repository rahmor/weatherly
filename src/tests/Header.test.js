import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from '../Header/Header';

describe('DailyWeather component', () => {
  it('should rendeer UI as expected', () => {
    const wrapper = shallow(<Header />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
