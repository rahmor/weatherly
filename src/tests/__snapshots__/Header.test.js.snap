// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyWeather component should rendeer UI as expected 1`] = `
<header
  className="Header-container"
>
  <h1
    className="Header-title"
  >
    Weatherly
  </h1>
  <form>
    <input
      placeholder="Search by City"
      type="search"
    />
    <button
      type="submit"
    >
      Search
    </button>
  </form>
  <h2>
    Atlanta, Ga
  </h2>
</header>
`;
